src/contexts/globalLayout/GlobalNavbar/CollectionSaveButtonWithCaption.tsx

```diff
@@ -0,0 +1,139 @@
+import { useCallback, useMemo, useState } from 'react';
+import styled from 'styled-components';
+
+import { Button } from '~/components/core/Button/Button';
+import colors from '~/components/core/colors';
+import { HStack, VStack } from '~/components/core/Spacer/Stack';
+import { BaseM, TitleDiatypeL } from '~/components/core/Text/Text';
+import { TextAreaWithCharCount } from '~/components/core/TextArea/TextArea';
+import transitions, {
+  ANIMATED_COMPONENT_TRANSLATION_PIXELS_SMALL,
+} from '~/components/core/transitions';
+import CloseIcon from '~/icons/CloseIcon';
+
+type Props = {
+  disabled?: boolean;
+  hasUnsavedChange?: boolean;
+  label?: string;
+  onSave: (caption: string) => void;
+};
+
+export function CollectionSaveButtonWithCaption({
+  disabled,
+  hasUnsavedChange,
+  label = 'Save',
+  onSave,
+}: Props) {
+  const [isShowPopup, setIsShowPopup] = useState(false);
+  const [caption, setCaption] = useState('');
+
+  const toggleSaveButton = () => {
+    setIsShowPopup(!isShowPopup);
+  };
```

Comment:
`useCallback` and `setIsShowPopup(previous => !previous);`

Reply:
Updated!